test/e2e/secrets_test.go

```diff
@@ -181,9 +182,36 @@ func testPrivateKeyChange(t *testing.T) {
 	err = tc.validateUpgradeableCondition(meta.ConditionTrue)
 	require.NoError(t, err, "operator Upgradeable condition not in proper state")
 
-	// Re-create the known private key so SSH connection can be re-established
-	// TODO: Remove dependency on this secret by rotating keys as part of https://issues.redhat.com/browse/WINC-655
-	require.NoError(t, tc.createPrivateKeySecret(true), "error confirming known private key secret exists")
+	// revert key changes so the test suite is able to SSH into the VM
+	require.NoError(t, tc.revertPrivateKeyChange())
+}
+
+// revertPrivateKeyChange changes the cloud private key to the known key if necessary, and waits for Machine nodes
+// to make use of it
+func (tc *testContext) revertPrivateKeyChange() error {
+	currentKey, err := tc.getCloudPrivateKey()
+	if err != nil {
+		return fmt.Errorf("error getting cloud private key: %w", err)
+	}
+	knownKey, err := os.ReadFile(gc.privateKeyPath)
```

Comment:
The most expensive operation here is reading the file and is happening twice for the worst-case scenario (when there is a mismatch). I suggest you rename this function as `ensureKnownPrivateKeySecret()` and make the adjustments to avoid the comparison and minimize the I/O operations while just resetting the known key and waiting for the expected status of the nodes.

Reply:
I'd rather not prematurely optimize here given this is a test.
Its important to make this check so that this function can be called safely in the case that the private key is already set.

 ensureKnownPrivateKeySecret() is not possible as the machine node refresh is the important bit here. thats not happening if the key wasnt changed.